src/test/ui/unsafe/rfc-2585-unsafe_op_in_unsafe_fn.rs

```diff
@@ -0,0 +1,22 @@
+#![feature(unsafe_block_in_unsafe_fn)]
+#![deny(unsafe_op_in_unsafe_fn)]
+#![deny(unused_unsafe)]
+
+unsafe fn unsf() {}
+
+unsafe fn foo() {
+    unsf();
+    //~^ ERROR call to unsafe function is unsafe and requires unsafe block
+    //~| WARNING it will become a hard error in a future release
```

Comment:
I don't think we have decided yet that it will become a hard error in a future release -- so for now, this should not be a future-compat lint, but just a normal lint, and allow-by-default on top of that.